File: ObjectiveC/PaytabsIntegrationObjectiveC/Controllers/PaymentDetailController.m
//
//  PaymentDetailController.m
//  PaytabsIntegrationObjectiveC
//
//  Created by gipl on 11/06/20.
//  Copyright © 2020 gipl. All rights reserved.
//

#import "PaymentDetailController.h"
#import "DetailModel.h"

@interface PaymentDetailController ()

@end

@implementation PaymentDetailController

- (void)viewDidLoad {
    [super viewDidLoad];
    self.tableView.tableFooterView = [[UIView alloc] initWithFrame:CGRectZero];
    self.title = @"Payment Detail";
}

//MARK:- TableView delegate and datasource methods
- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section{
    return [self.detailArray count];
}


- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
    
    UITableViewCell *cell = cell = [[UITableViewCell alloc] initWithStyle:UITableViewCellStyleValue1 reuseIdentifier: @"cell"];;
    cell.selectionStyle = UITableViewCellSelectionStyleNone;
    cell.detailTextLabel.numberOfLines = 0;
    
    DetailModel *dict = [self.detailArray objectAtIndex:indexPath.row]; //*******Retrive data from detail model
    
    cell.textLabel.text = dict.title;
    cell.detailTextLabel.text = dict.value;
    return cell;
}

@end

File: Swift/PayTabsIntegrationSwift/Controllers/PaymentDetailController.swift


import UIKit

class PaymentDetailController: UIViewController {

    var detailData = [CommonModel]()
    
    @IBOutlet weak var tableView: UITableView!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        self.title = "Payment Detail"
        tableView.tableFooterView = UIView()
    }
}

extension PaymentDetailController: UITableViewDelegate, UITableViewDataSource {
    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return detailData.count
    }
    
    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        let cell = UITableViewCell(style: .value1, reuseIdentifier: "cell")
        cell.selectionStyle = .none
        cell.detailTextLabel?.numberOfLines = 0
        
        let dict = detailData[indexPath.row]
        
        cell.textLabel?.text = dict.title
        cell.detailTextLabel?.text = dict.value
        
        return cell
    }
}
